diff --git a/setup.py b/setup.py
index 38e891c..e99fa9d 100644
--- a/setup.py
+++ b/setup.py
@@ -57,22 +57,12 @@ def get_package_version():
     with open(Path(current_dir) / 'deep_gemm' / '__init__.py', 'r') as f:
         version_match = re.search(r'^__version__\s*=\s*(.*)$', f.read(), re.MULTILINE)
     public_version = ast.literal_eval(version_match.group(1))
-
-    revision = ''
-    if DG_USE_LOCAL_VERSION:
-        # noinspection PyBroadException
-        try:
-            status_cmd = ['git', 'status', '--porcelain']
-            status_output = subprocess.check_output(status_cmd).decode('ascii').strip()
-            if status_output:
-                print(f'Warning: Git working directory is not clean. Uncommitted changes:\n{status_output}')
-                assert False, 'Git working directory is not clean'
-
-            cmd = ['git', 'rev-parse', '--short', 'HEAD']
-            revision = '+' + subprocess.check_output(cmd).decode('ascii').rstrip()
-        except:
-            revision = '+local'
-    return f'{public_version}{revision}'
+    # Auto-generate local version from CUDA and PyTorch versions.
+    torch_version = parse(torch.__version__)
+    cuda_version = torch.version.cuda
+    if cuda_version:
+        return f'{public_version}+cu.{cuda_version}.torch.{torch_version.major}.{torch_version.minor}'
+    return str(public_version)
 
 
 def get_platform():
@@ -182,6 +172,9 @@ class CachedWheelsCommand(_bdist_wheel):
             super().run()
 
 
+TORCH_VERSION = parse(torch.__version__)
+TORCH_MAJOR, TORCH_MINOR = TORCH_VERSION.major, TORCH_VERSION.minor
+
 if __name__ == '__main__':
     # noinspection PyTypeChecker
     setuptools.setup(
@@ -201,4 +194,5 @@ if __name__ == '__main__':
             'build_py': CustomBuildPy,
             'bdist_wheel': CachedWheelsCommand,
         },
+        install_requires=[f"torch=={TORCH_MAJOR}.{TORCH_MINOR}.*"]
     )
